presto-main/src/main/java/io/prestosql/util/DateTimeUtils.java

```diff
@@ -217,19 +196,35 @@ public static long parseTimestampWithoutTimeZone(String value)
         }
     }
 
+    public static long parseTimestampWithTimeZone(String value)
+    {
+        DateTime dateTime = TIMESTAMP_WITH_TIME_ZONE_FORMATTER.withOffsetParsed().parseDateTime(value);
+        return packDateTimeWithZone(dateTime);
+    }
+
     /**
-     * Parse a string (optionally containing a zone) as a value of TIMESTAMP type.
+     * Parse a string (optionally containing a zone) as a value of TIMESTAMP WITH TIME ZONE type.
      * If the string doesn't specify a zone, it is interpreted in {@code timeZoneKey} zone.
+     * <p>
+     * For example: {@code "2000-01-01 01:23:00"} is parsed to TIMESTAMP WITH TIME ZONE
+     * {@code 2000-01-01T01:23:00 <provided zone>} and {@code "2000-01-01 01:23:00 +01:23"}
+     * is parsed to TIMESTAMP WITH TIME ZONE {@code 2000-01-01T01:23:00.000+01:23}.
      *
-     * @return stack representation of legacy TIMESTAMP type
-     * @deprecated applicable in legacy timestamp semantics only
+     * @return stack representation of TIMESTAMP WITH TIME ZONE type
      */
-    @Deprecated
-    public static long parseTimestampWithoutTimeZone(TimeZoneKey timeZoneKey, String value)
+    public static long convertToTimestampWithTimeZone(TimeZoneKey timeZoneKey, String timestampWithTimeZone)
     {
-        return TIMESTAMP_WITH_OR_WITHOUT_TIME_ZONE_FORMATTER.withChronology(getChronology(timeZoneKey)).parseMillis(value);
+        DateTime dateTime = TIMESTAMP_WITH_OR_WITHOUT_TIME_ZONE_FORMATTER.withChronology(getChronology(timeZoneKey)).withOffsetParsed().parseDateTime(timestampWithTimeZone);
+        return packDateTimeWithZone(dateTime);
```

Comment:
The code here used to be known as `parseTimestampWithTimeZone(TimeZoneKey timeZoneKey, String timestampWithTimeZone)`, but in the diff it appears as a new code.

Reply:
Ah, that's because I inadvertently moved methods around. I'll restore the order to make the diff easier to review.